src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala

```diff
@@ -649,29 +669,34 @@ $DOCKERFILE_SIGIL
 
     // Generate the cache key file.
     val cacheKeyDestination = new File(imageDirectory, "conf/cacheKey.sha1")
-    val cacheKeyContents = {
-      // Hash the contents minus the local dependency jars and the cache key itself. For local
-      // dependency jars we use the git commit because it's more stable than the jar contents.
+    val cacheKeyContentsOption: Option[String] = {
+      // Hash the dependency image directly.
       val stableContentsHash = {
-        val ignoredFiles =
-          localDependencyMapping.map { case (_, destination) => destination }.toSet +
-            cacheKeyDestination
-        val allFiles = PathFinder(imageDirectory).***.filter { file =>
-          file.isFile && !ignoredFiles(file)
-        }.get
-        Utilities.hashFiles(allFiles, imageDirectory)
+        val dependencyDir = dependencyImageDir.value
+        Utilities.hashFiles(PathFinder(dependencyDir).***.filter(_.isFile).get, dependencyDir)
       }
 
-      // Find the git commits / random IDs for the current project and local dependencies.
-      val unstableContentsCommits = {
-        Def.taskDyn(gitCommitOrGuid.all(HelperDefs.dependencyFilter.value)).value :+
-          gitCommitOrGuid.value
+      // Find the git commits for the current project and local dependencies, since this is more
+      // stable than a hash of the contents.
+      val unstableContentsCommits: Seq[Option[String]] = {
+        (Def.taskDyn(gitCommitIfClean.all(HelperDefs.dependencyFilter.value)).value :+
```

Comment:
I really like that you're doing this by project, so an unrelated change won't dirty it.